src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -154,7 +103,7 @@ object FilterIndexRule
         val indexManager = Hyperspace
           .getContext(spark)
           .indexCollectionManager
-        val candidateIndexes = RuleUtils.getCandidateIndexes(indexManager, r)
+        val candidateIndexes = RuleUtils.getCandidateIndexes(spark, indexManager, r)
```

Comment:
can we just move the `indexManager` creation to `RuleUtils.getCandidateIndexes`? Since it's always used in conjunction?